File: app/src/main/java/com/example/englishapp/core/database/dao/ReviewHistoryDao.java
package com.example.englishapp.core.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.englishapp.core.database.entity.ReviewHistoryEntity;

import java.util.List;

@Dao
public interface ReviewHistoryDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertReviewHistory(ReviewHistoryEntity history);

    @Query("SELECT * FROM review_history WHERE vocabularyId = :vocabId ORDER BY reviewedAt DESC")
    LiveData<List<ReviewHistoryEntity>> getHistoryByVocabularyId(long vocabId);

    @Query("SELECT DISTINCT reviewedAt / 86400000 FROM review_history ORDER BY reviewedAt / 86400000 DESC")
    List<Long> getReviewedDaysDesc();
}

File: app/src/main/java/com/example/englishapp/core/database/dao/ReviewHistoryDao.kt
package com.example.englishapp.core.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.englishapp.core.database.entity.ReviewHistoryEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface ReviewHistoryDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertReviewHistory(history: ReviewHistoryEntity): Long

    @Query("SELECT * FROM review_history WHERE vocabularyId = :vocabId ORDER BY reviewedAt DESC")
    fun getHistoryByVocabularyId(vocabId: Long): Flow<List<ReviewHistoryEntity>>
}